src/m365/entra/commands/app/app-permission-list.ts

```diff
@@ -0,0 +1,264 @@
+        appId: typeof args.options.appId !== 'undefined',
+        appObjectId: typeof args.options.appObjectId !== 'undefined',
+        type: typeof args.options.type !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '-i, --appId [appId]' },
+      { option: '--appObjectId [appObjectId]' },
+      { option: '--type [type]', autocomplete: this.allowedTypes }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.appId && !validation.isValidGuid(args.options.appId as string)) {
+          return `${args.options.appId} is not a valid GUID`;
+        }
+
+        if (args.options.appObjectId && !validation.isValidGuid(args.options.appObjectId as string)) {
+          return `${args.options.appObjectId} is not a valid GUID`;
+        }
+
+        if (args.options.type && this.allowedTypes.map(x => x.toLowerCase()).indexOf(args.options.type.toLowerCase()) === -1) {
+          return `${args.options.type} is not a valid type. Allowed types are ${this.allowedTypes.join(', ')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['appId', 'appObjectId'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const appObjectId = await this.getAppObjectId(args.options);
+      const type = args.options.type ?? 'all';
+      const permissions = await this.getAppRegPermissions(appObjectId, type, logger);
+      await logger.log(permissions);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getAppObjectId(options: Options): Promise<string> {
+    if (options.appObjectId) {
+      return options.appObjectId;
+    }
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/myorganization/applications?$filter=appId eq '${formatting.encodeQueryParameter(options.appId!)}'&$select=id`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const res = await request.get<{ value: { id: string }[] }>(requestOptions);
+
+    if (res.value.length === 0) {
+      throw `No Azure AD application registration with ID ${options.appId} found`;
+    }
+
+    return res.value[0].id;
+  }
+
+  private async getAppRegPermissions(appObjectId: string, permissionType: string, logger: Logger): Promise<ApiPermission[]> {
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/v1.0/myorganization/applications/${appObjectId}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    const application = await request.get<Application>(requestOptions);
+
+    if ((application.requiredResourceAccess as RequiredResourceAccess[]).length === 0) {
```

Comment:
don't we do `(application.requiredResourceAccess as RequiredResourceAccess[])` like a couple of times? 
maybe it would increase readability if we will do it once and assign is to some var